compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2743,7 +2745,7 @@ impl<'tcx> Ty<'tcx> {
             | ty::Tuple(..) => (tcx.types.unit, false),
 
             ty::Str | ty::Slice(_) => (tcx.types.usize, false),
-            ty::Dynamic(..) => {
+            ty::Dynamic(_, _, DynKind::Dyn) => {
```

Comment:
This looks like a pretty serious bug in ptr_metadata_ty caused by Dyn and DynStar using the same type constructor.